### PR TITLE
networkd: introduce netdev "Netdevsim" Driver

### PR DESCRIPTION
**Pull request by ssahani**
This "netdevsim" as implied by the name is a tool for network developers and is a simulator.
This simulated networking device is used for testing various networking APIs and at this time
is particularly focused on testing hardware offloading related interfaces.

### CODE REVIEW

File: src/network/netdev/netdevsim.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (yuwata) [inline src/network/netdev/netdevsim.h:2]: missing SPDX license identifier.

Comment (ssahani) [reply src/network/netdev/netdevsim.h:2]: done

**Comment by poettering** [comment]
Humm, I am not convinced we should support stuff only useful for simulators in networkd proper. It's fine if people use that for their tests, but we shouldn't have to maintain that.

**Comment by yuwata** [comment]
I have not tested this, but I expect that this makes easier to debug `systemd-networkd`, right? The code added by this seems almost only definitions. So, I guess it seems not so hard to maintain this.

**Comment by ssahani** [comment]
yes actually we can test various APIs without real device for example https://patchwork.ozlabs.org/patch/843785/ . we have this on TODO but we can''t do this because I don;t have the device. Yes we just supporting a `kind=netdevsim`.

We have 
```
* networkd/udev: implement SR_IOV configuration in .link files:                                                                                                                                                     
  http://lists.freedesktop.org/archives/systemd-devel/2015-January/027451.html 
```
https://patchwork.ozlabs.org/patch/843785/

**Comment by poettering** [comment]
and the patch is trivial, hence let's merge this! thanks!

**Pull request merged**